src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/ClusterShardingLeavingSpec.cs

```diff
@@ -193,15 +193,7 @@ private void Join(RoleName from, RoleName to)
             {
                 Cluster.Join(Node(to).Address);
                 StartSharding();
-                Within(TimeSpan.FromSeconds(15), () =>
```

Comment:
This should fix it - the problem was that the fourth node never got to start its sharding system until the very end, so it was possible that a rebalance could happen after the `ShardLocations` snapshot was taken - which would corrupt the state of the test.

Better to let the entire cluster form all at once so the shard distribution can happen concurrently as the cluster forms, rather than staggering it. This way the sharding system has to do all of its redistribution once, rather than 3 times (once for each join + barrier in the previous code.)